src/gpu/nvidia/cudnn_matmul.hpp

```diff
@@ -116,24 +122,25 @@ struct cudnn_matmul_t : cudnn_matmul_base_t {
             }
             return true;
         }
+
+        std::shared_ptr<cublas_params> params_;
```

Comment:
Is there any reason why we want to have shared ownership over the params?

Reply:
It needs to be created in the primitive descriptor and it is used by the implementation, attempting to pass a unique_ptr back and forward from impl to primitive might not be desirable.